### PR TITLE
Fixing some slickgrid bugs

### PR DESCRIPTION
**Pull request by aasimkhan30**
1. Fixes the headers not scrolling with the table data when there are focusable elements in the header. 
1. Moves the anchor to be before the rows content  so the focus is first trapped in column headers and then the table data.

Fixes needed for https://github.com/microsoft/azuredatastudio/issues/19831

### CODE REVIEW

**Comment by alanrenmsft** [comment]
the changes look good, but I want to be extra cautious, please schedule a meeting with me and we can test these changes together.

File: slick.grid.js [hunk 1]
```diff
@@ -379,7 +381,7 @@ if (typeof Slick === "undefined") {
             //.on("click", handleClick)
             .on("scroll", handleScroll);
         $headerScroller
-            //.on("scroll", handleHeaderScroll)
+            .on("scroll", handleHeaderScroll)
```
Comment (Charles-Gagnon) [inline slick.grid.js:384]: Any idea why this was commented out? (git blame can help find out where that happened)

**Review by Charles-Gagnon — APPROVED** [review]

File: slick.grid.js (hunk 1 above)
Comment (aasimkhan30) [reply slick.grid.js:384]: I tried to find that out but it goes back to your initial commit 13 months ago

Comment (alanrenmsft) [reply slick.grid.js:384]: go to the previous repo and see if you can find anything.

Comment (alanrenmsft) [reply slick.grid.js:384]: https://github.com/anthonydresser/SlickGrid this is the original repo. found it by checking the history of package.json of ads.

**Comment by alanrenmsft** [comment]
as we discussed offline, there are other issues to be addressed, maybe you should mark this as draft for now.

**Comment by aasimkhan30** [comment]
@alanrenmsft, this PR now only contains the fixes necessary to close the accessibility issues.

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**